### ISSUE TITLE
Clippy Error during auto fix

### ISSUE THREAD

**Issue by Cryptex-github** [rust-lang/rust#99785]
Clippy reported error during auto fix

```
❯ cargo clippy --workspace --all-features --fix --allow-dirty --broken-code  -- -D clippy::all -D clippy::pedantic -D clippy::nursery -D clippy::cargo
    Checking ril v0.1.0 (C:\Users\alvin\OneDrive\Desktop\repos\ril)
warning: failed to automatically apply fixes suggested by rustc to crate `ril`

after fixes were automatically applied the compiler reported errors within these files:

  * src\encode.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0699]: the type of this value must be known to call a method on a raw pointer on it
  --> src\encode.rs:55:37
   |
55 |         unsafe { (data as *const _).cast::<T>().read() }
   |                                     ^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0699`.
Original diagnostics will follow.

error[E0699]: the type of this value must be known to call a method on a raw pointer on it
  --> src\encode.rs:55:37
   |
55 |         unsafe { (data as *const _).cast::<T>().read() }
   |                                     ^^^^

For more information about this error, try `rustc --explain E0699`.
error: could not compile `ril` due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `ril` due to previous error
```

Original code is
```rs
unsafe { (data as *const _ as *const T).read() }
```
Clippy attempted to change it to
```rs
unsafe { (data as *const _).cast::<T>().read() }
```
which leads to an error

Note: `data` is `&[u8]`
